src/main/java/com/cema/users/controllers/Controller.java

```diff
@@ -0,0 +1,163 @@
+
+import com.cema.users.domain.User;
+import com.cema.users.entities.CemaUser;
+import com.cema.users.exceptions.IncorrectCredentialsException;
+import com.cema.users.exceptions.UserExistsException;
+import com.cema.users.exceptions.UserNotFoundException;
+import com.cema.users.mapping.UserMapping;
+import com.cema.users.repositories.CemaUserRepository;
+import com.cema.users.services.login.LoginService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/v1")
+@Api(produces = "application/json", value = "Allows interaction with the users database and authorization operations. V1")
+@CrossOrigin(origins = "*")
+public class Controller {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Controller.class);
+
+    private final CemaUserRepository cemaUserRepository;
+    private final UserMapping userMapping;
+    private final LoginService loginService;
+
+    public Controller(CemaUserRepository cemaUserRepository, UserMapping userMapping, LoginService loginService) {
+        this.cemaUserRepository = cemaUserRepository;
+        this.userMapping = userMapping;
+        this.loginService = loginService;
+    }
+
+    @ApiOperation(value = "Retrieve a user data by username", response = User.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully found user"),
+            @ApiResponse(code = 404, message = "The user you were looking for is not found")
+    })
+    @GetMapping(value = "/users/{username}", produces = {MediaType.APPLICATION_JSON_VALUE})
+    public ResponseEntity<User> lookUpUser(
+            @ApiParam(
+                    value = "The username of the user we are looking for.",
+                    example = "merlinds")
+            @PathVariable("username") String userName) {
+
+        LOG.info("Request for user {}", userName);
+
+        CemaUser cemaUser = cemaUserRepository.findCemaUserByUserName(userName);
+        if (cemaUser == null) {
+            throw new UserNotFoundException(String.format("User %s doesn't exits", userName));
```

Comment:
"User %s doesn't exits" is repeated 3 times, maybe you could use a constant REF http://192.168.100.4:9000/project/issues?id=users-be&open=AXoREALpVXiR6bdbzjHb&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL

Reply:
good point